### DISCUSSION TITLE
Cancellationtoken info

### DISCUSSION THREAD

**Discussion by wimpie3** [jchristn/SuperSimpleTcp#47] [General]
Not a bug but I think more info is needed in the docs on what the cancellationtoken actually is. It's not used by the examples so I'm not even sure what it's purpose is. Is the entire send buffer cleared when this special "cancellation token" is added to the buffer?

**Comment by jchristn** [comment]
Hi @wimpie3 - the ```CancellationToken``` is ultimately passed to ```SendInternalAsync```, which passes it into ```stream.ReadAsync``` (from the input stream) and into ```NetworkStream.WriteAsync``` or ```SslStream.WriteAsync```, and, their respective ```FlushAsync```.  The documentation can be found here: https://docs.microsoft.com/en-us/dotnet/api/system.net.sockets.networkstream.readasync?view=net-5.0  Anecdotally (from Stack Overflow) and others, it appears that the token is only checked at the beginning of those calls (```ReadAsync```, ```WriteAsync```).

**Comment by wimpie3** [comment]
OK thanks.

**Reply by jchristn** [reply]
Moved to Discussion
